lib/bolt/pal/yaml_plan.rb

```diff
@@ -30,6 +30,12 @@ def initialize(name, plan)
           Parameter.new(param, definition)
         end.freeze
 
+        @private = plan['private']
+        unless @private.nil? || @private.is_a?(TrueClass) || @private.is_a?(FalseClass)
+          msg = "Plan #{@name} key 'private' must be a boolean, received: #{@private.inspect}"
+          raise Bolt::Error.new(msg, "bolt/invalid-plan")
+        end
```

Comment:
In `bolt plan show`, this error surfaces as a warning, but the plan will be marked as private. Not sure if this is desired behavior or not.

Reply:
It seems like this happens with other errors in YAML plans too - for example, if you don't have a `steps` key it will omit the plan but still return the plan show output (same behavior as invalid private value). I think that's technically fine, mostly because it's inline with existing behavior and if users try to run the plan or show it the error does get displayed. But it is a little unfortunate that it doesn't exactly encourage people to correct it since the error behavior is the same as the behavior they want :P